lsp/core/aws-lsp-core/src/content/contentRequestMiddleware.ts

```diff
@@ -0,0 +1,39 @@
+import { URI } from 'vscode-uri'
+
+/**
+ * Middleware pipeline handler for requesting content.
+ * Handlers are chained together so that a content requestor makes a single request call.
+ *
+ * Implementations have no knowledge of the calling pipeline, or of what other handlers are in the pipeline.
+ * An implementation may perform pre-processing on the request, and then call the delegate.
+ * An implementation may perform post-processing on the result returned by a delegate.
+ * If an implementation is capable of handling the request, it may return the result without calling the delegate.
```

Comment:
Doesn't this break the first assertion in this paragraph?  This implies that the handlers can make some assumption of the other downstream handlers, that they're not doing meaningful work.

For example, if there is a handler downstream that does some cross-cutting/bookkeeping concern, it would get skipped if the handler didn't continue passing the request down the pipeline even if it assumes it handled it completely.

Is there a way for an eng to immediately know what handlers can perform this short-circuiting behavior when composing the pipeline (without having to read through every handler's code)?

Reply:
The expectation is that one handler will ultimately produce the core response payload. Other handlers in the stack may handle a different aspect of the request relating to the core response payload.